src/server/pkg/hashtree/hashtree.go

```diff
@@ -463,7 +463,7 @@ func (h *hashtree) removeFromMap(path string) error {
 		delete(h.fs, path)
 	case unrecognized:
 		return errorf(Internal,
-			"malformed node at \"%s\": it's neither a file nor a directory", path)
+			"malformed file at \"%s\": it's neither a regular-file nor a directory", path)
```

Comment:
what does regular-file mean or imply? Is there another type of file? If not, just "file" is probably description enough, no?

Reply:
This is also why we were using the word "node" originally, which is also a very standard term. We need some way to refer to a general node in the filesystem which could be either a regular-file or a directory. The only 2 terms I know of for that are "node" and "file." "Node" is a more foreign word in general, but it might actually be more clear because the alternative "file," is more familiar but I think for most people it doesn't immediately have the association we want. It also necessitates the use of "regular-file" to distinguish what users normally think of as files from generic files which might be directories.

Reply:
That explains it. I was just trying to understand. basically "node" was meaning "file or directory" and now you're making it explicit with directory and regular-file. That seems fine. I've never referred to a dir as a file, but I understand why those are sort of interchangable in terms of paths. Keeping regular-file is fine -- i just wanted to understand it better. We could always just write out "file or directory" -- I'm not a perfect example audience in every case about which terms are meaningful to me. @dwhitena probably is the best judge.

Reply:
Writing out "file or directory" seems pretty lame. Does this message: "malformed file or directory at /foo/bar: it's neither a file nor a directory" seem clear to you? Referring to a dir as a file seems like a pretty natural aspect of the Pachyderm ontology to me. Afterall, most `-file` commands accept directories as arguments and return them as results. The exceptions being `put-file` and `get-file`. Is there a place we could explicitly spell out that `file` is a generic term? And that the specific examples of files are `regular-file`s and `directories`?